### PR TITLE
[GLUTEN-6561][CH] Fix incompatiable type exception throw in capture function while processing array literal with `transform`

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?

(Fixes: \#6561)

### CODE REVIEW

**Comment by taiyang-li** [comment]
Run Gluten Clickhouse CI

File: cpp-ch/local-engine/Parser/scalar_function_parser/CommonScalarFunctionParser.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ REGISTER_COMMON_SCALAR_FUNCTION_PARSER(ToUnixTimestamp, to_unix_timestamp, parse
 REGISTER_COMMON_SCALAR_FUNCTION_PARSER(Position, positive, identity);
 REGISTER_COMMON_SCALAR_FUNCTION_PARSER(Negative, negative, negate);
 REGISTER_COMMON_SCALAR_FUNCTION_PARSER(Pmod, pmod, pmod);
-REGISTER_COMMON_SCALAR_FUNCTION_PARSER(abs, abs, abs);
+REGISTER_COMMON_SCALAR_FUNCTION_PARSER(Abs, abs, abs);
```
Comment (baibaichen) [inline cpp-ch/local-engine/Parser/scalar_function_parser/CommonScalarFunctionParser.cpp:67]: is this modification related this issue or not?

**Comment by baibaichen** [comment]
@taiyang-li 

It's better to add more UTs.

File: cpp-ch/local-engine/Parser/scalar_function_parser/CommonScalarFunctionParser.cpp (hunk 1 above)
Comment (taiyang-li) [reply cpp-ch/local-engine/Parser/scalar_function_parser/CommonScalarFunctionParser.cpp:67]: 顺手改的

**Comment by baibaichen** [comment]
@taiyang-li why ignore test?

**Review by baibaichen — APPROVED** [review]
LGTM
